src/main/java/com/amazon/ion/impl/bin/WriteBuffer.java

```diff
@@ -1522,7 +1522,8 @@ public int writeFlexSym(int sid) {
             return writeFlexInt(sid);
         } else {
             writeByte(FlexInt.ZERO);
-            writeByte((byte) 0x90);
+            // The INLINE_SYMBOL_ZERO_LENGTH opcode is used for $0 in FlexSyms
+            writeByte(OpCodes.INLINE_SYMBOL_ZERO_LENGTH);
```

Comment:
🗺️ This (and the following change for `''`) should have been done as part of #800, but I missed it.